src/server/replica.cc

```diff
@@ -38,10 +38,15 @@ ABSL_FLAG(int, master_reconnect_timeout_ms, 1000,
           "Timeout for re-establishing connection to a replication master");
 ABSL_FLAG(bool, replica_partial_sync, true,
           "Use partial sync to reconnect when a replica connection is interrupted.");
-ABSL_FLAG(bool, replica_reconnect_on_master_restart, false,
-          "When in replica mode, and master restarts, break replication from master.");
+ABSL_FLAG(bool, break_replication_on_master_restart, false,
+          "When in replica mode, and master restarts, break replication from master to avoid "
+          "flushing the replica's data.");
 ABSL_DECLARE_FLAG(int32_t, port);
 
+// TODO: Remove this flag on release >= 1.22
+ABSL_RETIRED_FLAG(bool, replica_reconnect_on_master_restart, false,
```

Comment:
I agree that if this flag is set to true (not defaule) we should use this flag. We can remove it in 1.22 when we know that all prod instances are updated to use the new flag